pkg/module/disk_ref.go

```diff
@@ -0,0 +1,73 @@
+package module
+
+import (
+	"fmt"
+	"io/ioutil"
+
+	"github.com/gomods/athens/pkg/storage"
+	"github.com/pkg/errors"
+	"github.com/spf13/afero"
+)
+
+// DiskRef is a Ref implementation for modules on disk. It is not safe to use concurrently.
+//
+// Do not create this struct directly. use newDiskRef
+type diskRef struct {
+	root    string
+	fs      afero.Fs
+	version string
+	closer  func()
+}
+
+func newDiskRef(fs afero.Fs, root, version string) *diskRef {
+	return &diskRef{
+		fs:      fs,
+		root:    root,
+		version: version,
+		closer:  func() {},
+	}
+}
+
+// Clear is the Ref interface implementation. It deletes all module data from disk
+//
+// You should always call this function after you fetch a module into a DiskRef
+func (d *diskRef) Clear() error {
+	d.closer()
+	return d.fs.RemoveAll(d.root)
+}
+
+// read is the Ref interface implementation.
+func (d *diskRef) Read() (*storage.Version, error) {
+	ver := &storage.Version{}
```

Comment:
@robjloranger so would you prefer that I do `var ver *storage.Version` at the beginning of the function?

Reply:
No because that would still be a pointer type, I find it's more clear to read the return of a variable address in place of returning a variable that is not clearly a pointer. i.e. `var ver storage.Version` then on return `return &ver`

Reply:
ah, ok got it. I'll make that change